### ISSUE TITLE
j.l.Throwables.scala allocates 4Kish on stack.

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2282]
Noted in passing
```
 @noinline private[lang] def currentStackTrace(): Array[StackTraceElement] = {
    val cursor  = stackalloc[scala.Byte](2048.toUInt)
    val context = stackalloc[scala.Byte](2048.toUInt)

```
Allocating 4k (actually, slightly more) on stack is unusual in most cases
and sometimes is often considered un-friendly in multi-threaded
programs (one can hope that someday SN will get there).

Can this be done by heap allocation (possibly by pre-allocation, 
if doing heap allocation during a stack trace is likely to be a problem),.

Just an item to be considered either on some boring Summer afternoon
or as a case study in an Algorithms class.

**Comment by ebruchez** [comment]
This appears to have been addressed in `Throwables.scala`:

https://github.com/scala-native/scala-native/blob/0902082a86693fea8f689adab4a8867def226514/javalib/src/main/scala/java/lang/Throwables.scala#L41-L52

There is still stack allocation in the `test-interfaces` project:

https://github.com/scala-native/scala-native/blob/0902082a86693fea8f689adab4a8867def226514/test-interface/src/main/scala/scala/scalanative/testinterface/signalhandling/SignalConfig.scala#L79-L80

**Comment by ekrich** [comment]
Yes, it seems that this code needs to be unified and simplified somehow.

**Comment by ebruchez** [comment]
I don't understand enough to say, but there is a comment in `SignalConfig` about "using only async-signal-safe methods". If this is still valid, then might two different implementations be justified? Not sure why the runtime doesn't need "async-signal-safe methods" while the tests do.

**Comment by LeeTibbert** [comment]
I believe both sections of code need careful scrutiny. I seem to remember that it is being executed
as part of a signal handler.  By its nature, it is doing I/O. I suspect that I/O is not async-safe to begin
with, so how is a little extra error prone, possibly I/O, code like "alloca" going to hurt? ;-)

I suspect the ugly but necessary answer is pre-allocation at application start, of memory resident
pages (not paged in on demand).   Ugly, ugly, but double and triple faults on fault handling paths
are worse.

**Comment by WojciechMazur** [comment]
Memory is allocated on the heap now (once per thread)
